generate-release-cycle.py

```diff
@@ -0,0 +1,131 @@
+"""
+Read in a JSON and generate two CSVs and a Mermaid file
+"""
+from __future__ import annotations
+
+import csv
+import datetime as dt
+import json
+
+MERMAID_HEADER = """
+gantt
+    dateFormat  YYYY-MM-DD
+    title       Python release cycle
+    axisFormat  %Y
+""".lstrip()
+
+MERMAID_SECTION = """
+    section Python {cycle}
+    {release_status}               :{mermaid_status}   python{cycle}, {release_date},{eol}
+"""  # noqa: E501
+
+MERMAID_STATUS_MAPPING = {
+    "features": "",
+    "bugfix": "active,",
+    "security": "done,",
+    "end-of-life": "crit,",
+}
+
+CSV_HEADER = (
+    "Branch",
+    "Schedule",
+    "Status",
+    "First release",
+    "End-of-life",
+    "Release manager",
+)
+
+
+def cycle(branch: str) -> str:
+    """Convert branch name to version number"""
+    return "3.12" if branch == "main" else branch
```

Comment:
Wonder if we could have the info about current branch in the CSV somehow (bool column?) to avoid needing to remember to change the script.

Reply:
Or separate the version name from the branch name for the active branches, and have the key be the former rather than the latter? We could at least make that a script-level constant at the top.

Reply:
Have `"3.12"` in the JSON, and for the CSV replace the highest version with `main`?

Reply:
Sure, that should work too

Reply:
That could, though for programmatic use (i.e. what the JSON is primarily for) its likely to be helpful to have the branch name and version name as separate fields, since one or the other might be needed for a particular use.

Reply:
So something like?

```json
[
  {
    "branch": "main",
    "cycle": "3.12",
    ...
  },
  {
    "branch": "3.11",
    "cycle": "3.11",
    ...
  },
  {
    "branch": "3.12",
    "cycle": "3.12",
    ...
  },
...
```

Reply:
Yeah, that's what I was thinking, but just as a dict instead of a list of dicts with `cycle` also as the key, like we did for the `peps.json`, so its much simpler and more efficient to access a specific release cycle without having to iterate and check through the whole thing.

Reply:
Updated!